internal/provisioner/plain/controllers/bundleinstance_controller.go

```diff
@@ -343,20 +343,23 @@ func reconcileDesiredBundle(ctx context.Context, c client.Client, bi *rukpakv1al
 	}
 
 	// check whether there's an existing Bundle that matches the desired Bundle template
-	// specified in the BI resource, and if not, generate a new matches the template.
+	// specified in the BI resource, and if not, generate a new Bundle that matches the template.
 	b := util.CheckExistingBundlesMatchesTemplate(existingBundles, bi.Spec.Template)
 	if b == nil {
 		controllerRef := metav1.NewControllerRef(bi, bi.GroupVersionKind())
+		hash := util.GenerateTemplateHash(bi.Spec.Template)
```

Comment:
Should we add a comment here that explains why we're generating the template hash before building up the set of labels?

Reply:
Yes, that sounds good. Basically, my investigation found that that the hash is different on macOS versus linux, most likely due to the representation of the pointer being passed in on different platforms. The FNV hash itself is pretty straightforward and not likely to produce different results, so it must be the underlying data. 

Wherever we use the underlying hash library implementation, we just check for equality later on on-cluster (here and in OLM) which works for that specific os/arch combination. There are no tests that check "given an object, this is its hash" because those tests would seemingly produce different results on different platforms. 

So the end result is I pulled out the hardcoded hashes out of the tests, which watered down the value of the unit tests some, but enable them to pass on different platforms. The alternative would be to make these tests GOOS/GOARCH aware, which seems like overkill, since we have good feedback from the e2e regardless.